mi/dataset/parser/metbk_a_dcl.py

```diff
@@ -151,5 +123,85 @@ def __init__(self,
+                self.construct_instrument_particle_map(raw_data)
+                particle = self._extract_sample(particle_class,
+                                                None,
+                                                raw_data,
+                                                preferred_ts=DataParticleKey.PORT_TIMESTAMP)
+                self._record_buffer.append(particle)
+
+    def construct_instrument_particle_map(self, raw_data):
+        """
+        Breaks raw data list down into their particular particles.
+        Append particle and type into instrument particle map.
+        """
+        barometric_pressure             = raw_data[1]
+        relative_humidity               = raw_data[2]
+        air_temperature                 = raw_data[3]
+        longwave_irradiance             = raw_data[4]
+        precipitation                   = raw_data[5]
+        sea_surface_temperature         = raw_data[6]
+        sea_surface_conductivity        = raw_data[7]
+        shortwave_irradiance            = raw_data[8]
+        eastward_wind_velocity          = raw_data[9]
+        northward_wind_velocity         = raw_data[10]
+
+        global INSTRUMENT_PARTICLE_MAP
+        del INSTRUMENT_PARTICLE_MAP[:]      # Clear the list before making a new one
+        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(barometric_pressure, 'barometric_pressure', SENSOR_GROUP_BAROMETRIC_PRESSURE))
+        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(relative_humidity, 'relative_humidity', SENSOR_GROUP_RELATIVE_HUMIDITY))
+        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(air_temperature, 'air_temperature', SENSOR_GROUP_AIR_TEMPERATURE))
+        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(longwave_irradiance, 'longwave_irradiance', SENSOR_GROUP_LONGWAVE_IRRADIANCE))
+        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(precipitation, 'precipitation', SENSOR_GROUP_PRECIPITATION))
+        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(sea_surface_temperature, 'sea_surface_temperature', SENSOR_GROUP_SEA_SURFACE_TEMPERATURE))
+        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(sea_surface_conductivity, 'sea_surface_conductivity', SENSOR_GROUP_SEA_SURFACE_CONDUCTIVITY))
+        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(shortwave_irradiance, 'shortwave_irradiance', SENSOR_GROUP_SHORTWAVE_IRRADIANCE))
+        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(eastward_wind_velocity, 'eastward_wind_velocity', SENSOR_GROUP_EASTWARD_WIND_VELOCITY))
+        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(northward_wind_velocity, 'northward_wind_velocity', SENSOR_GROUP_NORTHWARD_WIND_VELOCITY))
+
+    @staticmethod
+    def add_particle(raw_data, particle, group):
+        """
+        This function will add particle to the instrument particle map
+        """
+        if isinstance(raw_data, float):
+            return particle, group, float
+        else:
```

Comment:
this function doesn't add a particle, it creates a particle tuple. Also, we don't want to create string types